src/main/java/graphql/validation/ArgumentValidationUtil.java

```diff
@@ -0,0 +1,84 @@
+package graphql.validation;
+
+import graphql.language.Argument;
+import graphql.language.ObjectField;
+import graphql.language.Value;
+import graphql.schema.GraphQLEnumType;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLScalarType;
+import graphql.schema.GraphQLType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+public class ArgumentValidationUtil extends ValidationUtil {
```

Comment:
@bbakerman I think a separate class is better because the whole context is surrounding a single argument. You would have to set the argument inside the validation util before doing validation and then clean it up afterwards. It couples validation to arguments and would make for less clean code. Just my 2c.